### DISCUSSION TITLE
Question: Sending email with custom Send Button

### DISCUSSION THREAD

**Discussion by alejofl** [InboxSDK/InboxSDK#1193] [Q&A] [answered]
Hey, I'm trying to replace the Gmail send button with another one, created by me.

I have the following code:

```Javascript
sdk.Compose.registerComposeViewHandler((composeView) => {
        const sendButton = document.createElement("div");
        sendButton.innerText = "Custom Send";
        sendButton.id = "sendButton";
        sendButton.onclick = async () => {
            console.log("Sending email");
            console.log(composeView.getSubject());
            composeView.send();
        }

        composeView.replaceSendButton({ el: sendButton });
});
```

Both console.logs are working, but the email is not sent. What am I doing wrong?

Any help is much appreciated. Thanks.

**Comment by wegry** [comment, marked as answer]
`ComposeView.prototype.send` works by simulating a click on Gmail's send button. If the button has been replaced, simulating a click won't work.

At Streak, we use `ComposeView.prototype.replaceSendButton` to make a button that when clicked, fires a request to our backend to send the message with Gmail's API.
